src/Compilers/CSharp/Test/Syntax/Parsing/RecordParsing.cs

```diff
@@ -897,5 +900,107 @@ public void WithParsing14()
             }
             EOF();
         }
+
+        [Fact]
+        public void WithParsing15()
```

Comment:
>WithParsing15 [](start = 20, length = 13)

Suggest you try `x with { A = e is T x B = x }`

Reply:
Good catch! I added a simple fix that may not be perfect but hopefully it will catch most of the cases one would hit in a simple demo